src/main/java/org/assertj/core/api/InstanceOfAssertFactories.java

```diff
@@ -0,0 +1,70 @@
+package org.assertj.core.api;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.DoublePredicate;
+import java.util.function.IntPredicate;
+import java.util.function.LongPredicate;
+import java.util.function.Predicate;
+
+/**
+ * Static {@link InstanceOfAssertFactory InstanceOfAssertFactories} for {@link Assert#instanceOf(InstanceOfAssertFactory)}.
+ *
+ * @since 3.13.0
+ */
+public final class InstanceOfAssertFactories {
```

Comment:
In order to make it easy for users to discover the factories they should be available in the entry point classes like `Assertions`, `BDDAssertions` and `WithAssertions`. 

To avoid the hassle of keeping the entry point classes in sync, we can simply make this class an interface and make the entry point classes to implement them.